internal/controller/controllers/bmh_agent_controller_test.go

```diff
@@ -842,8 +845,12 @@ var _ = Describe("bmac reconcile", func() {
 			})
 		})
 
-		Context("when bmh.Spec.Image exists", func() {
+		Context("when bmh.Spec.Image doesn't need to be updated", func() {
 			It("the detached annotation is not removed", func() {
+				host.Status.Provisioning.State = bmh_v1alpha1.StateProvisioned
+				host.Status.OperationHistory.Provision.Start = metav1.Time{Time: time.Now()}
```

Comment:
We don't really need to set this, this was relevant for the original implementation. I will remove it if there are changes requested on this PR.

Reply:
Now that you need to rebase anyway it's a perfect time to do it :sunglasses: